helper/common/common.go

```diff
@@ -358,6 +358,16 @@ func BigIntDivCeil(a, b *big.Int) *big.Int {
 		Div(result, b)
 }
 
+// SafeAddUint64 sums two unsigned int64 numbers if there are no overflow.
+// In case there is an overflow, it would return 0 and true, otherwise sum and false.
+func SafeAddUint64(a, b uint64) (uint64, bool) {
```

Comment:
Does it make sense to use atomic.AddUint64 here?

Reply:
If you are suggesting it because of thread safety, I don't feel we need it, because parameters are passed into the function by value and also they are not modified, and instead, a new local variable is returned, so in that context I don't see there should be any race conditions. Introducing of the `atomic` would make implementation a bit less readable IMO.